docs/src/rules/no-restricted-imports.md

```diff
@@ -230,6 +232,56 @@ import { AllowedObject as DisallowedObject } from "foo";
 
 :::
 
+#### allowImportNames
+
+Inverse of importNames. Restricts all imports from a module, except specified allowed ones.
+
+```json
+"no-restricted-imports": ["error", {
+  "paths": [{
+    "name": "import-foo",
+    "allowImportNames": ["Bar"],
+    "message": "Please use Bar from /import-bar/baz/ instead."
+  }]
+}]
+```
+
+Examples of **incorrect** code for `allowImportNames` in `paths`:
+
+Disallowing all import names except 'AllowedObject'.
+
+::: incorrect { "sourceType": "module" }
+
+```js
+/*eslint no-restricted-imports: ["error", { paths: [{
+    name: "foo",
+    allowImportNames: ["AllowedObject"],
+    message: "Only 'AllowedObject' is allowed to be imported from 'foo'."
+}]}]*/
+
+import { DisallowedObject } from "foo";
+```
+
+:::
+
+Examples of **correct** code for `allowImportNames` in `paths`:
+
+Disallowing all import names except 'AllowedObject'.
+
+::: correct { "sourceType": "module" }
+
+```js
+/*eslint no-restricted-imports: ["error", { paths: [{
+    name: "foo",
+    allowImportNames: ["AllowedObject"],
+    message: "Only 'AllowedObject' is allowed to be imported from 'foo'."
+}]}]*/
+
+import { AllowedObject } from "foo";
+```
+
+:::
+
 ### patterns
```

Comment:
can we add a section of `allowImportNames` option inside `patterns` too with suitable example this will assure the users that `patterns` too has this option.

Reply:
This has been added and commited.